ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -78,6 +82,24 @@
     SOME_NOT_ALL_DUPLICATE, // At least one of the message is a duplicate, but not all.
   }
 
+  private enum WriteState {
```

Comment:
would we need this if our delegate supported something like `getReplicaWriteState()`?

Reply:
Right now one delegate is created per AmbryServer and that delegate is shared with multiple blob stores (since it's many-to-one blob stores to disk managers, and many-to-one disk managers to storage managers).  We'd have to do something like pass down the ClusterParticipant from AmbryServer -> StorageManager -> DiskManager, and then DiskManager create a new Delegate per BlobStore each with the same ClusterParticipant reference... then you could have the Delegate save the WriteState instead of the BlobStore

Reply:
As discussed will add isSealed to ReplicaId in order to use underlying AmbryReplica isSealed() method and use ReplicaId for saving state instead of adding it to BlobStore